.config/doom/config.org
#+TITLE: Config
#+AUTHOR: David Petti

* TABLE OF CONTENTS :toc:
- [[#basic-doom][Basic DOOM]]
- [[#apheleia][Apheleia]]
- [[#cc][C/C++]]
- [[#latex][LaTeX]]
- [[#tramp][TRAMP]]
- [[#org][Org]]
  - [[#headers][Headers]]
  - [[#agenda][Agenda]]
  - [[#roam][Roam]]
- [[#typst][Typst]]

* Basic DOOM

#+begin_src emacs-lisp
(setq doom-theme 'doom-gruvbox)
(setq display-line-numbers-type 'relative)
(setq scroll-margin 8)
;; (doom/set-frame-opacity 60)
(setq +format-on-save-enabled-modes
      '(not emacs-lisp-mode sql-mode))
#+end_src

* Apheleia

#+begin_src emacs-lisp
(after! apheleia
  (setf (alist-get 'c-mode apheleia-mode-alist) 'clang-format
        (alist-get 'c++-mode apheleia-mode-alist) 'clang-format))

(setq apheleia-remote-algorithm 'remote)
#+end_src

* C/C++

#+begin_src emacs-lisp
(require 'dap-cpptools)
#+end_src

* LaTeX

#+begin_src emacs-lisp
(setq TeX-view-program-selection '((output-pdf "PDF Tools"))
      TeX-source-correlate-start-server t)
(add-hook 'TeX-after-compilation-finished-functions #'TeX-revert-document-buffer)

(map! :leader
      :desc "Latex Complie and view" "\ l" #'TeX-command-run-all)
#+end_src

* TRAMP

#+begin_src emacs-lisp
(setq tramp-default-method "ssh")
(setq tramp-verbose 1)

;; Disable version control checks on remote files
(setq vc-ignore-dir-regexp
      (format "\\(%s\\)\\|\\(%s\\)"
              vc-ignore-dir-regexp
              tramp-file-name-regexp))

;; Disable lock files remotely
(setq remote-file-name-inhibit-locks t)

;; Disable backups and autosaves for TRAMP
(add-to-list 'backup-directory-alist
             (cons tramp-file-name-regexp nil))
(add-to-list 'auto-save-file-name-transforms
             (list tramp-file-name-regexp nil t))

;; Use ControlMaster for SSH
(setq tramp-ssh-controlmaster-options
      "-o ControlMaster=auto -o ControlPersist=yes")
#+end_src

* Org

#+begin_src emacs-lisp
(after! org
  (setq org-directory "~/org")
)
#+end_src

** Headers

#+begin_src emacs-lisp
;; (custom-theme-set-faces!
;; 'doom-one
;; '(org-level-8 :inherit outline-3 :height 1.0)
;; '(org-level-7 :inherit outline-3 :height 1.0)
;; '(org-level-6 :inherit outline-3 :height 1.1)
;; '(org-level-5 :inherit outline-3 :height 1.2)
;; '(org-level-4 :inherit outline-3 :height 1.3)
;; '(org-level-3 :inherit outline-3 :height 1.4)
;; '(org-level-2 :inherit outline-2 :height 1.5)
;; '(org-level-1 :inherit outline-1 :height 1.6)
;; '(org-document-title  :height 1.8 :bold t :underline nil))
#+end_src

** Agenda

#+begin_src emacs-lisp
(after! org
  (setq org-agenda-files '("~/org/agenda.org"))

  (setq
   ;; org-fancy-priorities-list '("[A]" "[B]" "[C]")
   ;; org-fancy-priorities-list '("❗" "[B]" "[C]")
   ;; org-fancy-priorities-list '("🟥" "🟧" "🟨")
   org-priority-faces
   '((?A :foreground "#ff6c6b" :weight bold)
     (?B :foreground "#98be65" :weight bold)
     (?C :foreground "#c678dd" :weight bold))
   org-agenda-block-separator 8411)

  (setq org-agenda-custom-commands
        '(("v" "A better agenda view"
           ((tags "PRIORITY=\"A\""
                  ((org-agenda-skip-function '(org-agenda-skip-entry-if 'todo 'done))
                   (org-agenda-overriding-header "High-priority unfinished tasks:")))
            (tags "PRIORITY=\"B\""
                  ((org-agenda-skip-function '(org-agenda-skip-entry-if 'todo 'done))
                   (org-agenda-overriding-header "Medium-priority unfinished tasks:")))
            (tags "PRIORITY=\"C\""
                  ((org-agenda-skip-function '(org-agenda-skip-entry-if 'todo 'done))
                   (org-agenda-overriding-header "Low-priority unfinished tasks:")))
            (tags "customtag"
                  ((org-agenda-skip-function '(org-agenda-skip-entry-if 'todo 'done))
                   (org-agenda-overriding-header "Tasks marked with customtag:")))

            (agenda "")
            (alltodo ""))))))
#+end_src

** Roam

#+begin_src emacs-lisp
(after! org
  (setq org-roam-directory (file-truename "~/org/roam/"))
)
#+end_src

* Typst

#+begin_src emacs-lisp
(setq typst-preview-invert-colors "never")
#+end_src


.config/doom/config.el
(setq doom-theme 'doom-gruvbox)
(setq display-line-numbers-type 'relative)
(setq scroll-margin 8)
;; (doom/set-frame-opacity 60)

;; (after! ccls
;;   (setq ccls-initialization-options '(:index (:comments 2) :completion (:detailedLabel t))
;;         ccls-args '("-log-file=/tmp/ccls.log", "-std=c++23")
;; ;; -isystem/opt/homebrew/Cellar/gcc/15.1.0/bin/../lib/gcc/current/gcc/aarch64-apple-darwin24/15/../../../../../../include/c++/15 -isystem/opt/homebrew/Cellar/gcc/15.1.0/bin/../lib/gcc/current/gcc/aarch64-apple-darwin24/15/../../../../../../include/c++/15/aarch64-apple-darwin24 -isystem/opt/homebrew/Cellar/gcc/15.1.0/bin/../lib/gcc/current/gcc/aarch64-apple-darwin24/15/../../../../../../include/c++/15/backward -isystem/opt/homebrew/Cellar/gcc/15.1.0/bin/../lib/gcc/current/gcc/aarch64-apple-darwin24/15/include -isystem/opt/homebrew/Cellar/gcc/15.1.0/bin/../lib/gcc/current/gcc/aarch64-apple-darwin24/15/include-fixed -isystem/Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk/usr/include -isystem/Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk/System/Library/Frameworks")
;;         ccls-executable "ccls")
;;   (set-lsp-priority! 'ccls 0)) ; optional as ccls is the default in Doom

(setq TeX-view-program-selection '((output-pdf "PDF Tools"))
      TeX-source-correlate-start-server t)
(add-hook 'TeX-after-compilation-finished-functions #'TeX-revert-document-buffer)

(map! :leader
      :desc "Latex Complie and view" "\ l" #'TeX-command-run-all)

(after! org
  (setq org-directory "~/org")
)

(after! org
  (setq org-agenda-files '("~/org/agenda.org"))

  (setq
   ;; org-fancy-priorities-list '("[A]" "[B]" "[C]")
   ;; org-fancy-priorities-list '("❗" "[B]" "[C]")
   ;; org-fancy-priorities-list '("🟥" "🟧" "🟨")
   org-priority-faces
   '((?A :foreground "#ff6c6b" :weight bold)
     (?B :foreground "#98be65" :weight bold)
     (?C :foreground "#c678dd" :weight bold))
   org-agenda-block-separator 8411)

  (setq org-agenda-custom-commands
        '(("v" "A better agenda view"
           ((tags "PRIORITY=\"A\""
                  ((org-agenda-skip-function '(org-agenda-skip-entry-if 'todo 'done))
                   (org-agenda-overriding-header "High-priority unfinished tasks:")))
            (tags "PRIORITY=\"B\""
                  ((org-agenda-skip-function '(org-agenda-skip-entry-if 'todo 'done))
                   (org-agenda-overriding-header "Medium-priority unfinished tasks:")))
            (tags "PRIORITY=\"C\""
                  ((org-agenda-skip-function '(org-agenda-skip-entry-if 'todo 'done))
                   (org-agenda-overriding-header "Low-priority unfinished tasks:")))
            (tags "customtag"
                  ((org-agenda-skip-function '(org-agenda-skip-entry-if 'todo 'done))
                   (org-agenda-overriding-header "Tasks marked with customtag:")))

            (agenda "")
            (alltodo ""))))))

(after! org
  (setq org-roam-directory (file-truename "~/org/roam/"))
)

(setq typst-preview-invert-colors "never")
